cdap-app-templates/cdap-program-report/cdap-program-report-base/src/main/scala/co/cask/cdap/report/ReportGenerationHelper.scala

```diff
@@ -140,6 +145,47 @@ object ReportGenerationHelper {
     writeToFile(System.currentTimeMillis().toString, Constants.LocationName.SUCCESS_FILE, reportIdDir)
   }
 
+  /**
+    * From the filters in ReportGenerationRequest figure out namespaces if they are provided, get the start
+    * and end time range of query and use default for all other fields of report summary and write to summary file
+    * write count file with count 0 and write success file.
+    * @param request ReportGenerationRequest
+    * @param reportIdDir report directory where the summary, count and success files will be written to
+    */
+  private def writeEmptyFileSummaryCountAndSuccessFiles(request: ReportGenerationRequest,
+                                                        reportIdDir: Location) = {
+    val namespacesAggregates = getNamespaceAggregates(request)
+
+    val summary = new ReportSummary(namespacesAggregates, request.getStart, request.getEnd,
+      new ArrayBuffer[ArtifactAggregate](), new DurationStats(0l, 0l, 0.0),
+      new StartStats(0l, 0l), new ArrayBuffer[UserAggregate](), new ArrayBuffer[StartMethodAggregate]())
+    writeSummaryToFile(summary, reportIdDir)
+    val count = 0
+    // Create a _COUNT file and write the total number of report records in it
+    writeToFile(count.toString, Constants.LocationName.COUNT_FILE, reportIdDir)
```

Comment:
why do we need three separate files instead of a single file? Can't you just put the count and time in the summary?